vendors/nxp/boards/lpc54018iotmodule/CMakeLists.txt

```diff
@@ -0,0 +1,317 @@
+    "${lpc54018_dir}/utilities"
+    "${lpc54018_dir}/utilities/io"
+    "${lpc54018_dir}/utilities/log"
+    "${lpc54018_dir}/utilities/str"
+    "${lpc54018_dir}/CMSIS/Include"
+)
+
+# Kernel
+afr_mcu_port(kernel)
+
+target_sources(
+    AFR::kernel::mcu_port
+    INTERFACE
+        ${lpc54018_src}
+        ${drivers_src}
+        ${mcuxpresso_src}
+        ${usb_device_src}
+        ${usb_osa_src}
+        ${wifi_qca_src}
+        ${utilities_src}
+        "${AFR_KERNEL_DIR}/portable/GCC/ARM_CM4F/port.c"
+        "${AFR_KERNEL_DIR}/portable/GCC/ARM_CM4F/portmacro.h"
+        "${AFR_KERNEL_DIR}/portable/MemMang/heap_4.c"
+)
+
+set(kernel_inc_dirs
+    ${lpc54018_includes}
+    "${AFR_KERNEL_DIR}/portable/GCC/ARM_CM4F"
+    "${AFR_KERNEL_DIR}/portable/CCS/ARM_CM3"
+    "${lpc54018_app_dir}/config_files"
+    "${lpc54018_app_dir}/application_code/nxp_code"
+    "$<IF:${AFR_IS_TESTING},${AFR_TESTS_DIR},${AFR_DEMOS_DIR}>/include"
+
+)
+
+target_include_directories(
+    AFR::kernel::mcu_port
+    INTERFACE $<$<NOT:$<COMPILE_LANGUAGE:ASM>>:${kernel_inc_dirs}>
+)
+
+
+# WiFi
+afr_mcu_port(wifi)
+target_sources(
+    AFR::wifi::mcu_port
+    INTERFACE "${lpc54018_ports_dir}/wifi/aws_wifi.c"
+)
+
+# PKCS11
+afr_mcu_port(pkcs11)
+target_sources(
+    AFR::pkcs11::mcu_port
+    INTERFACE 
+        "${lpc54018_ports_dir}/pkcs11/aws_pkcs11_pal.c"
+        "${lpc54018_ports_dir}/pkcs11/hw_poll.c"
+        "${AFR_MODULES_PORTS_DIR}/pkcs11/mbedtls/aws_pkcs11_mbedtls.c"
+)
+
+# Link to AFR::pkcs11_mbedtls if you want to use default implementation based on mbedtls.
+target_link_libraries(
+    AFR::pkcs11::mcu_port
+    INTERFACE 
+        3rdparty::mbedtls
+        AFR::crypto
+)
+
+# Secure sockets
+afr_mcu_port(secure_sockets)
+target_link_libraries(
+    AFR::secure_sockets::mcu_port
+    INTERFACE 
+        AFR::tls
+        AFR::wifi
+)
+
+target_sources(
+    AFR::secure_sockets::mcu_port
+    INTERFACE "${lpc54018_ports_dir}/secure_sockets/aws_secure_sockets.c"
+)
+
+# -------------------------------------------------------------------------------------------------
+# Amazon FreeRTOS demos and tests
+# -------------------------------------------------------------------------------------------------
+file(COPY "${lpc54018_dir}/mcuxpresso/aws_demos_Debug.ld" DESTINATION . )
+file(COPY "${lpc54018_dir}/mcuxpresso/aws_demos_Debug_memory.ld" DESTINATION . )
+file(COPY "${lpc54018_dir}/mcuxpresso/aws_demos_Debug_library.ld" DESTINATION . )
+
+set(CMAKE_EXECUTABLE_SUFFIX ".axf")
+
+if(AFR_IS_TESTING)
+    set(exe_target aws_tests)
+else()
+    set(exe_target aws_demos)
+endif()
+
+afr_glob_src(board_code_src DIRECTORY "${lpc54018_app_dir}/application_code/nxp_code")
+afr_glob_src(config_files_src DIRECTORY "${lpc54018_app_dir}/config_files")
+
+add_executable(${exe_target}
+    ${board_code_src}
+    ${config_files_src}
+    "${lpc54018_app_dir}/application_code/main.c"
+    # It is unknown why hw_poll.c has to be included here
```

Comment:
So if it's compiled into pkcs11 static library, the linker cannot find the function symbols defined in it?

Reply:
Yeah...

amazon-freertos/modules/libraries/3rdparty/mbedtls/library/entropy.c:120: undefined reference to `mbedtls_hardware_poll'